static/js/stream_popover.js

```diff
@@ -469,8 +471,21 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
             },
         );
     }
-
+    function set_stream_topic_typeahead() {
+        const topic_input = $(".inline_topic_edit");
+        const new_stream_id = Number(stream_widget.value(), 10);
+        const new_stream_name = sub_store.get(new_stream_id).name;
+        if (new_stream_name !== current_stream_name) {
```

Comment:
Do we need this conditional? It may be cleaner to not have it, and I think it is incorrect as written, since we don't mutate `current_stream_name`.